src/poweredup-browser.ts

```diff
@@ -132,31 +132,29 @@ export class PoweredUP extends EventEmitter {
             let buf: Buffer = Buffer.alloc(0);
             device.subscribeToCharacteristic(Consts.BLECharacteristic.LPF2_ALL, (data: Buffer) => {
                 buf = Buffer.concat([buf, data]);
-                const len = buf[0];
-                if (len >= buf.length) {
+                while (buf[0] <= buf.length) {
+                    const len = buf[0];
                     const message = buf.slice(0, len);
                     buf = buf.slice(len);
                     if (message[2] === 0x01 && message[3] === 0x0b) {
-                        process.nextTick(() => {
```

Comment:
This is already in an event callback, I don't see the point of the `process.nextTick()`

Reply:
My memory is fading me on this one, but I think this was in place as advertisement packets (with manufacturer data to determine hub type) came in slightly after the discovery, and this was in place to allow it to be processed already. If I recall correctly this was particularly a problem with the Boost Move Hub.

It may be it's not required since the v6 rearchitecting, but I'll need to do some solid testing with different hub types before I feel comfortable removing it.